src/app/AlwaysOn.Shared/TelemetryExtensions/AppInsightsRequestHandler.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.Azure.Cosmos;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace AlwaysOn.Shared.TelemetryExtensions
+{
+    public class AppInsightsRequestHandler : RequestHandler
+    {
+        public AppInsightsRequestHandler(TelemetryClient telemetryClient)
+        {
+            _telemetryClient = telemetryClient;
+        }
+
+        private readonly TelemetryClient _telemetryClient;
+
+        public override async Task<ResponseMessage> SendAsync(RequestMessage request, CancellationToken cancellationToken)
+        {
+            var title = request.Properties.ContainsKey("Operation") ? request.Properties["Operation"].ToString() : $"{request.Method} {request.RequestUri.OriginalString}";
+            var dbClientEndpoint = request.Properties.ContainsKey("DbClientEndpoint") ? request.Properties["DbClientEndpoint"].ToString() : null;
+
+            using var dependency = _telemetryClient.StartOperation<DependencyTelemetry>(title);
+
+            //
+            // Comparison with the original telemetry object:
+            //
+            //var telemetry = new DependencyTelemetry()
+            //{
+            //    OK - Type = AppInsightsDependencyType,
+            //    OK - Data = $"ObjectId={objectId}, Partitionkey={partitionKey}",
+            //    OK - Name = $"Delete {typeof(T).Name}",
+            //    OK - Timestamp = startTime,
+            //    OK - Duration = diagnostics != null ? diagnostics.GetClientElapsedTime() : overallDuration,
+            //    OK - Target = diagnostics != null ? diagnostics.GetContactedRegions().FirstOrDefault().uri?.Host : _dbClient.Endpoint.Host,
+            //    REMOVED - Success = success
+            //};
+            //
+            //if (response != null)
+            //    OK - telemetry.Metrics.Add("CosmosDbRequestUnits", response.RequestCharge);
+
+            var response = await base.SendAsync(request, cancellationToken);
+
+            dependency.Telemetry.Type = "Azure DocumentDB";
+            dependency.Telemetry.Data = request.RequestUri.OriginalString; // Will be shown as "Command" in Application Insights
+            dependency.Telemetry.Target = response.Diagnostics != null ? response.Diagnostics.GetContactedRegions().FirstOrDefault().uri?.Host : dbClientEndpoint;
+            dependency.Telemetry.ResultCode = ((int)response.StatusCode).ToString();
+            dependency.Telemetry.Success = response.IsSuccessStatusCode;
```

Comment:
Is a 404 still a success?

Reply:
It's not (as described in the PR description ;) ). I don't remember why we made it a succes, because it shouldn't be - trying to get a non existent object is suspicious.

Reply:
Hmm I'm not sure I would count a 404 as a non-success. Try to GET item /123 which just doesnt exist (anymore?) is still a successful operation against cosmos. I would have said not success=false means something actually went wrong

Reply:
But it is actually wrong. The client is trying to get something which is not there.

But fine - I could simply create special case just for 404 in the handler. The other one was Conflict. Do we consider that a success too?